### ISSUE TITLE
Date To Unix Timestamp

### ISSUE THREAD

**Issue by zhudock** [OraOpenSource/oos-utils#18]
Honestly not sure how many would benefit from this, but here it is. Note this version includes processing the local timezone

```
CREATE OR REPLACE FUNCTION date_to_unix (in_date DATE)
   RETURN NUMBER
IS
   l_epoch   NUMBER;
BEGIN
   SELECT   (in_date - TO_DATE ('19700101', 'yyyymmdd')) * 86400
          - TO_NUMBER (SUBSTR (TZ_OFFSET (SESSIONTIMEZONE), 1, 3)) * 3600
     INTO l_epoch
     FROM DUAL;

   RETURN l_epoch;
END date_to_unix;
```

**Comment by jrwarwick** [comment]
I'm glad to see this suggested, zhudock. How about the inverse? I can imagine some ETL scenarios where both would be useful.

**Comment by martindsouza** [comment]
@jrwarwick and @zhudock should we call this get epoch date? or something similar? It isn't necessarily unix date.

**Comment by zhudock** [comment]
epoch date sounds good to me

**Comment by zhudock** [comment]
Here is the epoch_to_date function as well

```
CREATE OR REPLACE FUNCTION epoch_to_date (in_epoch NUMBER)
   RETURN DATE
IS
   l_date   DATE;
BEGIN
   SELECT   TO_DATE ('19700101', 'yyyymmdd')
          + ((in_epoch + (TO_NUMBER(SUBSTR(TZ_OFFSET(SESSIONTIMEZONE), 1, 3)) * 3600)) / 86400)
     INTO l_date
     FROM DUAL;

   RETURN l_date;
END epoch_to_date;
```

**Comment by commi235** [comment]
Maybe reduce this done to a "one-liner":
```
  return DATE '1970-01-01'
       + ((in_epoch + (TO_NUMBER(SUBSTR(TZ_OFFSET(SESSIONTIMEZONE), 1, 3)) * 3600)) / 86400);
```
The select from dual isn't necessary in this case.

**Comment by martindsouza** [comment]
@zhudock @commi235 @jrwarwick I've called these functions `date2epoch` and `epoch2date`. Initially I was hoping to call it `get_epoch_date` but couldn't think of an appropriate `get_date > return epoch` name.

When I check the date in Linux (`date` and `date +%s`) I needed to add a `+1` to the `tz_offset` for the numbers to match the OS.

I'll mark this as closed but feel free to leave any comments about the names of the modifications I made.
